### ISSUE TITLE
JSON output of metrics

### ISSUE THREAD

**Issue by prayerslayer** [siimon/prom-client#22]
Hi, I'm trying to write a status page for my web app with the help of this library. I use it via a custom [Koa](https://github.com/koajs/koa) middleware that collects request metrics. Now it would be useful if I could get these metrics as JSON in order to render it nicely in a static page. Would you accept a PR for this?

**Comment by siimon** [comment]
Not sure it makes much sense to expose the raw data, but I see no downside adding it so sure if you open a PR for it I'll be happy to merge it.

**Comment by vot** [comment]
My scenario is slightly different but I also need the same thing (to ensure I'm not trying to register the same metric twice from different files in order to avoid duplication).

Just raised a PR: https://github.com/siimon/prom-client/pull/23

**Comment by siimon** [comment]
Thanks! @vot your use case is reasonable, but maybe it should be solved in the library itself so we don't allow metrics with the same name twice. 

But before doing that change we should probably check how other Prometheus libraries handles this

**Comment by vot** [comment]
Agreed @siimon, I didn't want to go too deep into the lib with this kind of changes as it might not be desired. For instance what if you want to register two different types of metrics with the same name? (not that I think this is wise but still it's a possibility). I completely agree that checking the official libraries should be the way to go here, will have a look if I have a moment to spare.

In the meantime I just put myself a simple lib in my app that implements `prom-client` with a simple object I put references to metrics I register in order to keep track of what's already there. Might need to update this when the PR gets merged.

As a side note I think it would be useful to have `type` stored as a property of a metric object. At one point when using the `metrics` array I ran a `_.find` from Lodash, only to realise I can't match on type without using `instanceof` to determine what kind of metric I've got on my hands.
That's just me though trying to put few things together quickly and probably won't affect many people.

PS Thanks for putting this lib together, saved me a lot of time trying to figure out Prometheus :)

**Comment by siimon** [comment]
Just made a quick test on the python client and it seems to allow duplicate names so i guess we should follow that convention.

About the type property, should maybe be in it's own ticket - but I don't see why you want to enumerate metrics at all :) In my opinion they should be more like write only and only exposed through the /metrics endpoint in a supported Prometheus text format. Everything else is probably feature creep of this library. But exposing the metrics as JSON is something that could be useful but I'm not sure we should do anything more complicated than that!

**Comment by vot** [comment]
Agreed, we probably shouldn't be making assumptions about what the users may want to do.

I suppose my case is very specific - in addition to what I mentioned earlier I'll probably need to aggregate the metrics from few different workers in a cluster, just to make things more interesting. Will let you know if I need any more changes :)

**Comment by siimon** [comment]
Closing this one, @vot provided a getMetricsAsJSON function - published to npm with version 3.4.7

**Comment by shubhendumadhukar** [comment]
Can this be done on the AggegratorRegistry as well? Currently example shows following
```
const metrics = await aggregatorRegistry.clusterMetrics();
res.set("Content-Type", aggregatorRegistry.contentType);
res.send(metrics);
```
I was wondering if I can get these metrics in a json format.
